src/Std/Time/Date/Unit/Day.lean

```diff
@@ -0,0 +1,251 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.Time
+
+namespace Std
+namespace Time
+namespace Day
+open Lean Internal
+
+set_option linter.all true
+
+/--
+`Ordinal` represents a bounded value for days, which ranges between 1 and 31.
+-/
+def Ordinal := Bounded.LE 1 31
+  deriving Repr, BEq, LE, LT
+
+instance : ToString Ordinal where
+  toString x := toString x.val
+
+instance : OfNat Ordinal n :=
+  inferInstanceAs (OfNat (Bounded.LE 1 (1 + (30 : Nat))) n)
+
+instance {x y : Ordinal} : Decidable (x ≤ y) :=
+  inferInstanceAs (Decidable (x.val ≤ y.val))
+
+instance {x y : Ordinal} : Decidable (x < y) :=
+  inferInstanceAs (Decidable (x.val < y.val))
+
+instance : Inhabited Ordinal where default := 1
+
+/--
+`Offset` represents an offset in days. It is defined as an `Int` with a base unit of 86400
+(the number of seconds in a day).
+-/
+def Offset : Type := UnitVal 86400
+  deriving Repr, BEq, Inhabited, Add, Sub, Mul, Div, Neg, LE, LT, ToString
+
+instance : OfNat Offset n := ⟨UnitVal.ofNat n⟩
+
+instance {x y : Offset} : Decidable (x ≤ y) :=
+  inferInstanceAs (Decidable (x.val ≤ y.val))
+
+instance {x y : Offset} : Decidable (x < y) :=
+  inferInstanceAs (Decidable (x.val < y.val))
+
+/--
+`Period` is an enumeration representing different times of the day : morning, afternoon, evening, and night.
+-/
+inductive Period
+  /-- Represents the morning period. -/
+  | morning
+
+  /-- Represents the afternoon period. -/
+  | afternoon
+
+  /-- Represents the evening period. -/
+  | evening
+
+  /-- Represents the night period. -/
+  | night
+  deriving Repr, BEq, Inhabited
+
+namespace Period
+
+/--
+Determines the `Period` of the day based on the given hour.
+
+- If the hour is between 20 and 4, it returns `night`.
+- If the hour is between 17 and 20, it returns `evening`.
+- If the hour is between 12 and 17, it returns `afternoon`.
+- If the hour is between 5 and 12, it reutrns `morning`.
+-/
+@[inline]
+def fromHour (hour : Hour.Ordinal) : Day.Period :=
+  if hour ≥ 20 ∨ hour ≤ 4 then
+    .night
+  else if hour ≥ 17 then
+    .evening
+  else if hour ≥ 12 then
+    .afternoon
+  else
+    .morning
+
+end Period
```

Comment:
This seems very subjective (both to individuals and to different cultures) to me; does it really belong in the standard library?

Reply:
I'm going to remove it.

Reply:
I'm surprised I can't even find a specification for that format in Java, besides the list you paste above!